File: java/SendSMS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import org.apache.http.Header;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.auth.AuthScope;
import org.apache.http.auth.UsernamePasswordCredentials;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.DefaultHttpClient;

/* XXX didn't use org.json to be simple
import org.json.JSONObject; 
*/

public final class SendSMS {
    public static void main(String[] args) {
        String url = "https://api.bluehouselab.com/smscenter/v1.0/sendsms";
        DefaultHttpClient client = new DefaultHttpClient();
        try {
            // basic authentication
            client.getCredentialsProvider().setCredentials(
                AuthScope.ANY,
                new UsernamePasswordCredentials(Config.appid, Config.apikey)
            );
            HttpPost httpPost = new HttpPost(url);
            httpPost.setHeader("Content-type", "application/json; charset=utf-8");
            String json = "{\"sender\":\""+Config.sender+"\",\"receivers\":[\""+Config.receiver+"\"],\"content\":\""+Config.content+"\"}";
            StringEntity se = new StringEntity(json);
            httpPost.setEntity(se);

            HttpResponse httpResponse = client.execute(httpPost);
            System.out.println(httpResponse.getStatusLine().getStatusCode());

            InputStream inputStream = httpResponse.getEntity().getContent();
            if(inputStream != null) {
                BufferedReader bufferedReader = new BufferedReader( new InputStreamReader(inputStream));
                String line = "";
                while((line = bufferedReader.readLine()) != null)
                    System.out.println(line);
                inputStream.close();
            }
        } catch (Exception e) {
            System.err.println("Error: "+e.getLocalizedMessage());
        } finally {
            client.getConnectionManager().shutdown();
        }

    }
}

File: c#/sendsms.cs
using System;
using System.Net;

class SendSMS
{
    static void Main(string[] args)
    {
        string url = "https://api.bluehouselab.com/smscenter/v1.0/sendsms";
        string appid = "example";
        string apikey = "c5dd7e7dkjp27377l903c42c032b413b";

        string json = @"{
            ""sender"": ""01000000000"",
            ""receivers"": [""01000000000""],
            ""content"": ""나는 유리를 먹을 수 있어요. 그래도 아프지 않아요""}";

        WebClient client = new WebClient();
        NetworkCredential creds = new NetworkCredential(appid, apikey);
        client.Credentials = creds;
        client.Headers[HttpRequestHeader.ContentType] = "application/json; charset=utf-8";

        try
        {
            string response = client.UploadString(url, json);
            Console.WriteLine(response);
        }
        catch (WebException e)
        {
            HttpStatusCode status = ((HttpWebResponse)e.Response).StatusCode;
            Console.WriteLine("{0}", (int)status);
            Console.WriteLine("{0}", status.ToString());
        }
    }
}
